test/wasm2asm/i32.2asm.js

```diff
@@ -188,13 +188,6 @@ function asmFunc(global, env, buffer) {
   return x >>> 0 >= y >>> 0 | 0;
  }
  
```

Comment:
looks like there are only wasm2asm tests here, please add direct tests for the changes to the passes as well. (that might actually help me see why the second iteration of remove-non-js-ops is needed.)

Reply:
Sure! I added that in https://github.com/WebAssembly/binaryen/pull/1563/commits/cefc31ae8825640cd72504f92fbf47d36864793d